### PR TITLE
Adding support for MaxSessionDuration and allowing sts:TagSession

### PR DESCRIPTION
**Pull request by JeffAshton**

### CODE REVIEW

**Comment by JeffAshton** [comment]
FYI @AntonBazhal

File: README.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ provider:
               - 222222222222 # principal as accountId
               - 'arn:aws:iam::333333333333:root' # principal as ARN
               - Fn::Import: cloudformation-output-arn-2 # principal as CloudFormation Output Value ARN
+            allowTagSession: True # can optionally be defined to include sts:TagSession in assume role policy
```
Comment (AntonBazhal) [inline README.md:27]: Is there any harm in just always adding it?

File: add-permissions.js [hunk 2]
```diff
@@ -66,7 +66,7 @@ module.exports = class AwsAddLambdaAccountPermissions {
         }
 
         if (role) {
-          [].concat(role).forEach(({ name, principals }) => {
+          [].concat(role).forEach(({ allowTagSession, maxSessionDuration, name, principals }) => {
```
Comment (AntonBazhal) [inline add-permissions.js]:
```suggestion
          [].concat(role).forEach(({ allowTagSession, maxSessionDuration = 3600, name, principals }) => {
```

File: add-permissions.js [hunk 3]
```diff
@@ -80,6 +80,11 @@ module.exports = class AwsAddLambdaAccountPermissions {
               throw new Error(`Roles must have unique names [${name}]`);
             }
 
+            let stsAction = 'sts:AssumeRole';
+            if (allowTagSession) {
+              stsAction = ['sts:AssumeRole', 'sts:TagSession'];
+            }
```
Comment (AntonBazhal) [inline add-permissions.js:86]: If we decide to keep `allowTagSession`
```suggestion
            const stsAction = ['sts:AssumeRole'];
            if (allowTagSession) {
              stsAction.push('sts:TagSession');
            }
```

Comment (JeffAshton) [reply add-permissions.js:86]: Lol, I wrote that orginally, but then switched to reduce drift in existing tests.

File: add-permissions.js (hunk 2 above)
Comment (JeffAshton) [reply add-permissions.js]: Does that work in node 8.x?

https://github.com/rschick/serverless-plugin-lambda-account-access/blob/9e15aa34f2987d59d7324423bdf8bc8c963efc1e/package.json#L5

File: README.md (hunk 1 above)
Comment (JeffAshton) [reply README.md:27]: I was avoiding breaking changes.  I personally don't care.  :)

File: add-permissions.js (hunk 2 above)
Comment (JeffAshton) [reply add-permissions.js]: It does!!

File: README.md (hunk 1 above)
Comment (AntonBazhal) [reply README.md:27]: It's not breaking as it's adding an action, not removing it. Plus, that action should probably always be there.

Comment (JeffAshton) [reply README.md:27]: I would generally agree, but from a least privilege standpoint it's a gray area.  I'm good with whatever you gentlemen decide.  I'm just interested in switching the lms to 12 hours and tagging.

File: add-permissions.js (hunk 3 above)
Comment (AntonBazhal) [reply add-permissions.js:86]: If that's the goal, I'd use a ternary below, like
```js
{
  Action: allowTagSession
    ? ['sts:AssumeRole', 'sts:TagSession']
    : 'sts:AssumeRole'
}
```

File: README.md (hunk 1 above)
Comment (AntonBazhal) [reply README.md:27]: I'm OK with leaving it as is

File: add-permissions.js (hunk 3 above)
Comment (JeffAshton) [reply add-permissions.js:86]: Sure. Accomplishes the same result.  I've never personally been a fan of that operator.  Seen it abused too many times.

**Review by AntonBazhal — APPROVED** [review]

**Pull request merged**